back/src/modules/products/dto/products.input.ts

```diff
@@ -84,4 +94,16 @@ export class UpdateProductInput {
   @IsObject()
   @Field(() => ProductCharacteristicsInput, { nullable: true })
   characteristics: ProductCharacteristicsInput;
+
+  @IsOptional()
+  @IsNotEmpty()
+  @IsNumber()
+  @Field(() => Float)
+  price?: number;
+
+  @IsOptional()
+  @IsNotEmpty()
+  @IsNumber()
+  @Field(() => Float)
+  percentDiscount?: number;
```

Comment:
**Suggestion:** The 'price' and 'percentDiscount' fields are marked as optional but also have the @IsNotEmpty() decorator. This is contradictory and could lead to validation issues.
```suggestion
  @IsOptional()
  @IsNumber()
  @Field(() => Float)
  price?: number;

  @IsOptional()
  @IsNumber()
  @Field(() => Float)
  percentDiscount?: number;
```